### PR TITLE
Remove unnecessary masked_fill in deberta models

### LINKED ISSUE #35162: Did you mean masked_fill_ in modeling_deberta.py?
### System Info

- `transformers` version: 4.47.0
- Platform: Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.24.6
- Safetensors version: 0.4.4
- Accelerate version: 1.0.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.6.0.dev20241205+cu124 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: <fill in>
- GPU type: NVIDIA GeForce RTX 4060 Laptop GPU

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

masked_fill is called but the result is not used: https://github.com/huggingface/transformers/blob/main/src/transformers/models/deberta/modeling_deberta.py#L293.

The same issue happens in modeling_deberta_v2.py.

### Expected behavior

Use masked_fill_ instead.

### PR DESCRIPTION
**Pull request by xadupre**
# What does this PR do?

Both models Deberta and DebertaV2 executes one instruction with no impact on the result. It is the last line among the 

```python
attention_scores = attention_scores.masked_fill(~(attention_mask), torch.finfo(query_layer.dtype).min)
# bsz x height x length x dimension
attention_probs = nn.functional.softmax(attention_scores, dim=-1)
attention_probs.masked_fill(attention_mask, 0)  # no impact on what follows
```

The line can be removed safely. If we replace it by one inplace modification ``attention_probs.masked_fill(attention_mask, 0)`` then the attention_probs will be null because:

* line 1: replace all False mask values by - infinity
* line 2: compute softmax, then the replaced values become 0
* line 3: if modified with inplace modification, it would replace all True mask value by 0

At the end, the output tensor would be null. No unit test was added because the numerical results are not impacted.

Fixes #35162.

### CODE REVIEW

**Review by Rocketknight1 — APPROVED** [review]
This analysis makes sense, yes. I suspect the original intent was that the line would actually do `_masked_fill(~attention_mask, 0)`, which would ensure that all masked positions were fully set to 0, and didn't accidentally survive the first mask step as a tiny but non-zero value.

I agree that the line does nothing, and that the precaution is unneeded though, so thank you for the PR!
